model/shared/src/main/scala/hmda/model/institution/Institution.scala

```diff
@@ -1,40 +1,38 @@
 package hmda.model.institution
 
+import hmda.model.institution.Agency.UndeterminedAgency
+import hmda.model.institution.ExternalIdType.UndeterminedExternalId
+import hmda.model.institution.InstitutionType.UndeterminedInstitutionType
+
 /**
  * A financial institution, geared towards requirements for filing HMDA data.
  */
 case class Institution(
-    id: String,
-    name: String,
-    externalIds: Set[ExternalId],
-    agency: Agency,
-    institutionType: InstitutionType,
-    hasParent: Boolean,
-    cra: Boolean = false // TODO do we have this info when creating the institution? if so, then don't default here.
-) extends Serializable {
-
-  private val extIdsByType: Map[ExternalIdType, ExternalId] = externalIds.map(extId => (extId.idType, extId)).toMap
-
-  /**
-   * Derives the respondentId for a given Institution based on [[hmda.model.institution.Agency]] and [[hmda.model.institution.InstitutionType]],
-   * the rules for which can be found in section "1.4 - Respondent Identification Numbers for 2017 HMDA Filers" of the
-   * <a href="http://www.consumerfinance.gov/data-research/hmda/static/for-filers/2017/2017-HMDA-File-Specifications.pdf">2017 HMDA File Specifications</a>
-   */
-  def respondentId: Either[InvalidRespondentId, ExternalId] = {
-
-    agency.externalIdTypes.get(institutionType.depositoryType) match {
-      case None => Left(UnsupportedDepositoryTypeByAgency(id, agency, institutionType.depositoryType))
-      case Some(extIdType) =>
-
-        extIdsByType.get(extIdType) match {
-          case None => Left(RequiredExternalIdNotPresent(id, extIdType))
-          case Some(extId) => Right(ExternalId(extId.id, extIdType))
-        }
-    }
-  }
-
+  id: String,
+  agency: Agency,
+  activityYear: Int,
+  institutionType: InstitutionType,
+  cra: Boolean,
+  externalIds: Set[ExternalId],
+  emailDomains: EmailDomains,
+  respondent: Respondent,
+  hmdaFilerFlag: Boolean,
+  parent: Parent,
+  assets: Int,
+  otherLenderCode: Int,
+  topHolder: TopHolder
+)
+case object Institution {
+  def empty: Institution = Institution("", UndeterminedAgency, -1, UndeterminedInstitutionType, cra = false, Set(),
+    EmailDomains("", "", ""), Respondent(ExternalId("", UndeterminedExternalId), "", "", "", ""), hmdaFilerFlag = false,
+    Parent("", -1, "", "", ""), -1, -1, TopHolder(-1, "", "", "", ""))
 }
 
+case class EmailDomains(email2015: String, email2014: String, email2013: String)
```

Comment:
So, it looks like David did this based off of Debbie's work.  @hkeeler, what will you need in terms of email domains?  I'm assuming we'll need future years of emails, so it would probably be best to change this data structure... will you only ever need the previous three years?  Do you need to know which year the email is from, or can I just return a list of all emails for an institution?